Code/Framework/AzCore/AzCore/Settings/CommandLineParser.h

```diff
@@ -0,0 +1,173 @@
+            //! the "project-cache-path" does have its value set to "<value>"
+            NextTokenIsValueIfNonOption,
+            //! This indicates that the option is a flag
+            //! and contains no additional value parameters
+            //! For a token sequence of `--force <value>`
+            //! The "force" option uses a value of "". This allows the user to treated is a flag option
+            //! The "<value>" token in this case is treated as positional parameter
+            IsFlag,
+            // This indicates the the next token should be parsed as the value in all cases
+            // Even if the next token is an option (--foo)
+            // So a token sequence of `-project-path --project-cache-path <value>`
+            // will have the "project-path" option value set to "--project-cache-path"
+            // The "<value>" token would then be parsed as a positional parameter
+            NextTokenIsValue
+        };
+        //! Determines the default action the command line should take when it parses an option token WITHOUT a value
+        //! @param option string containing the name of the parsed command line option
+        //! @return The default action to take for the option.
+        //! The default return value is to treat the next command line token as the value for the option string
+        static OptionAction DefaultOptionAction(AZStd::string_view option);
+
+        //! Callback function which is invoked when an option token by itself is parsed
+        //! @return The type of action to take for the option, which is to either parse the next command line token
+        //! as the value for the option or treat the option as just a CLI flag, such as `--quiet` or `-verbose`
+        using OptionActionFunc = AZStd::function<OptionAction(AZStd::string_view option)>;
+
+        //! Function invoked ONLY after parsing an option Token that was not split from a CLI delimiter such as an '='
+        //! The algorithm that determines when this function is called is explained with the following sample command line arguments
+        //! `Editor --project-path=<project-path> --force --engine-path <engine-path> positionalArgument
+        //! For the "project-path" option, this function is NOT invoked since the value is gathered by spliting the CLI token on '='
+        //! result: option="project-path", value="<project-path>"
+        //! For the "force" option, this function IS invoked to determine how the next token should be treated
+        //! If the function returns `OptionAction::NextTokenIsValueIfNonOption`, then the following "--engine-path" token is read as an
+        //! option If the function returns `OptionAction::IsFlag`, then the following "--engine-path" token is read as an option If the
+        //! function returns `OptionAction::NextTokenIsValue`, then the following "--engine-path" token is treated as the value for the
+        //! "force" option
```

Comment:
Suggestion: Reformat the comments so that the start of each If sentence is a new line.
```suggestion
        //! If the function returns `OptionAction::NextTokenIsValueIfNonOption`, then the following "--engine-path" token is read as an
        //! option
        //! If the function returns `OptionAction::IsFlag`, then the following "--engine-path" token is read as an option
        //! If the function returns `OptionAction::NextTokenIsValue`, then the following "--engine-path" token is treated
        //! as the value for the "force" option.
```